torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1021,6 +1021,38 @@ Tensor masked_scatter_backward(const Tensor & grad, const Tensor & mask, IntArra
   return mask_selected.view(sizes);
 }
 
+std::tuple<Tensor, Tensor, Tensor> onboarding_attn_backward(const variable_list& grads, Tensor& q, Tensor& k, Tensor& v){
+  auto grad_o = grads[0];
+  auto grad_a = grads[1];
+
+  if (!grad_o.defined()){
+    grad_o = at::zeros({q.size(0), v.size(1)}, q.options());
```

Comment:
This is kind of cheating :D
Can you make this function more efficient by not materializing these?